script/child/DeployQWAaveV2.s.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: Apache 2
+pragma solidity 0.8.23;
+
+import {DeployBase} from '../helpers/DeployBase.sol';
+
+import {QWManager} from 'contracts/QWManager.sol';
+import {QWRegistry} from 'contracts/QWRegistry.sol';
+import {QWAaveV2} from 'contracts/child/QWAaveV2.sol';
+import {Script} from 'forge-std/Script.sol';
+
+contract DeployQWAaveV2 is Script, DeployBase {
+  struct ConfigParams {
+    address aaveLendingPool;
+  }
+
+  QWAaveV2 public qwAaveV2;
+  QWRegistry public qwRegistry;
+  QWManager public qwManager;
+
+  error InvalidAddress(); // Error for invalid registry address
+
+  function run() public {
+    vm.startBroadcast();
+
+    // Sanity check deployment parameters.
+    DeploymentBaseParams memory baseParams = _readBaseEnvVariables();
+    ConfigParams memory configParams = _readEnvVariables();
+
+    qwManager = QWManager(baseParams.qwManager);
```

Comment:
we should read the contracts from deployment directory first as well, this way env could be optional for deployed contracts.
Though priority should env > deployed contracts

Reply:
do you mean, create an additional deployment directory and deployer adds the addresses after deployment
and then it'd be used next time for fetching the addresses
PS: env files will be posted on github (sepolia.env, amoy.env will be there on github), only public variables are there in them

Reply:
once we deploy the contracts, the contracts are available in deployment directory. Similar to hardhat we can fetch the addresses from there. That's the suggestion.

Reply:
We will need to export the deployed addresses anyways so this just creates feature request.